CYCLONEDDS_QUALITY_DECLARATION.md

```diff
@@ -1,150 +1,210 @@
+
+Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format and start at the first character following the PATCH that is not a decimal digit.
+
+See also the [Releases section](https://www.eclipse.org/projects/handbook/#release) of the Eclipse project handbook which discusses _Major_, _Minor_, and _Service_ release criteria (where _Service_ corresponds to PATCH in the above description.
 
 ### Version Stability [1.ii]
 
-The latest release of `CycloneDDS` is considered stable, but there is no documentation about whether all future releases should be considered stable.
+Cyclone DDS is at a stable version. The current version is the highest versioned Git tag that consists of three decimal numbers separated by periods. For each minor release, there exists a releases/MAJOR.MINOR.x branch that is used for PATCH releases.
+
+The change history can be found in the [CHANGELOG](https://github.com/eclipse-cyclonedds/cyclonedds/blob/master/CHANGELOG.rst).
 
 ### Public API Declaration [1.iii]
 
-The public API for `CycloneDDS` is made up of symbols defined in the header files it installs.
-In the source repository, these header files reside with each of the modules that make up `CycloneDDS`.
+Symbols starting with `dds_` or `DDS_` and that are accessible after including only the top-level `dds.h` file, unless explicitly stated otherwise in a documentation comment, constitute the Public API.
+
+Definitions available in installed include files that are outside this definitions are either internal or part of an Evolving API that is expected to eventually become part of the Public API. The distinction between Evolving APIs and internal definitions is currently not well-defined.
+
+In the source repository, these header files reside with the modules that make up Cyclone DDS.
 
 ### API Stability Policy [1.iv]
 
-There is no published policy for API stability in `CycloneDDS`, however there is some information in the [Releases section](https://www.eclipse.org/projects/handbook/#release) of the Eclipse project handbook which states that releases which include breaking API changes are considered "Major".
+Cyclone DDS provides Public API stability for PATCH and MINOR releases. Cyclone DDS strives to provide Public API stability for MAJOR releases. Cyclone DDS does not guarantee Public API stability for MAJOR releases.
+
+Cyclone DDS provides Evolving API stability for PATCH releases and strives to provide Evolving API stability for MINOR releases. Cyclone DDS does not guarantee Evolving API stability for MINOR and MAJOR releases.
+
+The RMW interface is what ROS 2 deals with, and for a variety of reasons it may be decided to rely on unstable interfaces (or even implementation details), to better support ROS 2's design decisions that do not fit so well with DDS. Given ROS 2's importance to Cyclone DDS the stable API will slowly to support all, but without a defined time-line for that to happen.
```

Comment:
```suggestion
The RMW interface is what ROS 2 deals with, and for a variety of reasons it may be decided to rely on unstable interfaces (or even implementation details), to better support ROS 2's design decisions that do not fit so well with DDS. Given ROS 2's importance to Cyclone DDS the Public API is expected to eventually cover all of ROS 2's needs, but without a defined time-line for that to happen.
```